backends/openweather.go

```diff
@@ -0,0 +1,200 @@
+	flag.BoolVar(&ow.debug, "openweather-debug", false, "openweather backend: print raw requests and responses")
+}
+
+func (ow *openWeatherConfig) fetch(url string) (*openWeatherResponse, error) {
+	res, err := http.Get(url)
+	if ow.debug {
+		fmt.Printf("Fetching for %s \n", url)
+	}
+	if err != nil {
+		return nil, fmt.Errorf(" Unable to get (%s) %v", url, err)
+	}
+	defer res.Body.Close()
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to read response body (%s): %v", url, err)
+	}
+
+	if ow.debug {
+		fmt.Printf("Response (%s) %s", url, string(body))
+	}
+
+	var resp openWeatherResponse
+	if err = json.Unmarshal(body, &resp); err != nil {
+		return nil, fmt.Errorf("Unable to unmarshal response (%s): %v\nThe json body is: %s", url, err, string(body))
+	}
+	return &resp, nil
+}
+
+func (ow *openWeatherConfig) parseDaily(dataInfo []listBlock, numdays int) []iface.Day {
+	var openWeather []iface.Day
+	var day *iface.Day
+
+	for _, data := range dataInfo {
+		slot, err := ow.parseCond(data)
+		if err != nil {
+			log.Println("Error parsing hourly weather condition:", err)
+			continue
+		}
+		if day == nil {
+			day = new(iface.Day)
+			day.Date = slot.Time
+		}
+		if day.Date.Day() == slot.Time.Day() {
+			day.Slots = append(day.Slots, slot)
+		}
+		if day.Date.Day() != slot.Time.Day() {
+			openWeather = append(openWeather, *day)
+			if len(openWeather) >= numdays {
+				break
+			}
+			day = new(iface.Day)
+			day.Date = slot.Time
+			day.Slots = append(day.Slots, slot)
+		}
+
+	}
+	return openWeather
+}
+
+func (ow *openWeatherConfig) parseCond(dataInfo listBlock) (iface.Cond, error) {
+	var ret iface.Cond
+	codemap := map[int]iface.WeatherCode{
+		500: iface.CodeLightRain,
+		501: iface.CodeHeavyRain,
+		502: iface.CodeHeavyRain,
+		503: iface.CodeHeavyRain,
+		800: iface.CodeSunny,
+		802: iface.CodePartlyCloudy,
+		803: iface.CodePartlyCloudy,
+		804: iface.CodePartlyCloudy,
+	}
+	ret.Code = iface.CodeUnknown
+	ret.Desc = dataInfo.Weather[0].Main
+	ret.Humidity = &(dataInfo.Main.Humidity)
+	ret.TempC = &(dataInfo.Main.Temp)
+	if &dataInfo.Wind.Deg != nil {
+		p := int(dataInfo.Wind.Deg)
+		ret.WinddirDegree = &p
+	}
+	if &(dataInfo.Wind.Speed) != nil && (dataInfo.Wind.Speed) > 0 {
+		windSpeed := (dataInfo.Wind.Speed * 3.6)
+		ret.WindspeedKmph = &(windSpeed)
+	}
+	if val, ok := codemap[dataInfo.Weather[0].ID]; ok {
+		ret.Code = val
+	}
+
+	if &dataInfo.Rain.MM3h != nil {
+		mmh := (dataInfo.Rain.MM3h / 1000) / 3
+		ret.PrecipM = &mmh
+	}
+
+	ret.Time = time.Unix(dataInfo.Dt, 0)
+
+	return ret, nil
+}
+
+func (ow *openWeatherConfig) Fetch(location string, numdays int) iface.Data {
+	var ret iface.Data
+	s := strings.Split(location, ",")
+
+	lat, lon := s[0], s[1]
+
+	resp, err := ow.fetch(fmt.Sprintf(openweatherUri, lat, lon, ow.apiKey, ow.lang))
+	if err != nil {
+		log.Println("Cant fetch today")
+	}
+	ret.Current, err = ow.parseCond(resp.List[0])
```

Comment:
This fails if the `resp` return is nil. Use `log.Fatalf()` previously (also including the returned `err`) to abort the program before trying to access `resp.List`.